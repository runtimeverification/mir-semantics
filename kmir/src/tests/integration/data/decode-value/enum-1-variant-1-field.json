{
  "bytes": [
    255,
    0,
    0,
    0
  ],
  "types": [
    [
      0,
      {
        "PrimitiveType": {
          "Int": "I32"
        }
      }
    ]
  ],
  "typeInfo": {
    "EnumType": {
      "name": "Foo",
      "adt_def": 1,
      "discriminants": [
        0
      ],
      "fields": [
        [
          0
        ]
      ],
      "layout": {
        "fields": {
          "Arbitrary": {
            "offsets": [
              {
                "num_bits": 0
              }
            ]
          }
        },
        "variants": {
          "Single": {
            "index": 0
          }
        },
        "abi": {
          "Scalar": {
            "Initialized": {
              "value": {
                "Int": {
                  "length": "I32",
                  "signed": true
                }
              },
              "valid_range": {
                "start": 0,
                "end": 4294967295
              }
            }
          }
        },
        "abi_align": 4,
        "size": {
          "num_bits": 32
        }
      }
    }
  }
}
