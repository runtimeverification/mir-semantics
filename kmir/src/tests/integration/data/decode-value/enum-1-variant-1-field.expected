Aggregate ( variantIdx ( 0 ) , Integer ( 255 , 4 , true ) .List )
